File: src/Coduniverse.Domain/SpaceObject.cs
namespace Coduniverse.Domain
{
    public class SpaceObject : BaseEntity
    {
        public string Name { get; set; }
        public int Age { get; set; }
        public SpaceObjectType Type { get; set; }
        public StarSystem StarSystem { get; set; }
        public int Diameter { get; set; }
        public int Mass { get; set; }
    }
}

File: src/Coduniverse.Client/src/app/models/spaceobject.ts
import { StarSystem } from "./star-system";

export class Spaceobject {
    constructor(
        public id: string,
        public name: string,
        public age: number,
        public type: number,
        public typeName: string,
        public starSystemId: string,
        public mass: number
    ) { }
}
